src/datepicker/stateful-calendar.js

```diff
@@ -10,18 +10,21 @@ import StatefulContainer from './stateful-container.js';
 import Calendar from './calendar.js';
 import type {CalendarPropsT, StatefulDatepickerPropsT} from './types.js';
 
-function StatefulComponent(props: StatefulDatepickerPropsT<CalendarPropsT>) {
-  return (
-    <StatefulContainer {...props}>
-      {extendedProps => <Calendar {...extendedProps} />}
-    </StatefulContainer>
-  );
-}
+type PropsT<T> = StatefulDatepickerPropsT<CalendarPropsT<T>>;
 
-StatefulComponent.defaultProps = {
-  initialState: {},
-  stateReducer: (type, nextState) => nextState,
-  onSelect: () => {},
-};
+class StatefulComponent<T = Date> extends React.Component<PropsT<T>> {
```

Comment:
There seems to be a bug in prettier where passing a default in a generic type(ie. `<T = Date>`) breaks its parser.  There happened to only be two simple components that used to be functional components that needed to have default generics passed to them, and so the easiest short term solution was to swap them over to classes.  Ideally that bug gets resolved and we can switch back.